app/models/user.rb

```diff
@@ -493,6 +493,10 @@ def results_team?
     group_member?(UserGroup.teams_committees_group_wrt)
   end
 
+  def senior_results_team?
+    active_roles.any? { |role| role.group == UserGroup.teams_committees_group_wrt && role.metadata.at_least_senior_member? }
```

Comment:
Can you realise this using an ActiveRecord query instead of loading the roles into memory?

Reply:
Done